### PR TITLE
feat: replace legacy Activity table

### PR DESCRIPTION
**Pull request by luchaos**

### CODE REVIEW

File: app/Helpers/database/ticket.php [hunk 1]
```diff
@@ -15,22 +14,24 @@
...
     }
 
-    $value = getUserActivityRange($user, $firstLogin, $lastLogin)
-        && time() - strtotime($firstLogin) > 86400 // 86400 seconds = 1 day
-        && getRecentlyPlayedGames($user, 0, 1, $userInfo)
+    $user = User::where('User', $username)->first();
+
+    // TODO test this
+    $value = $user->Created->diffInDays() > 1
+        && getRecentlyPlayedGames($username, 0, 1, $userInfo)
```
Comment (luchaos) [inline app/Helpers/database/ticket.php:34]: Open TODO

File: app/Helpers/database/user-activity.php [hunk 2]
```diff
@@ -1,110 +1,27 @@
...
 use App\Site\Models\User;
 use App\Support\Cache\CacheKey;
-use Carbon\Carbon;
 use Illuminate\Support\Facades\Cache;
 
 /**
  * @deprecated see WriteUserActivity listener
  */
-function postActivity(string|User $userIn, int $type, ?int $data = null, ?int $data2 = null): bool
+function updateActivity(int $activityID): void
```
Comment (luchaos) [inline app/Helpers/database/user-activity.php:17]: still needed?

File: app/Helpers/database/user-activity.php [hunk 3]
```diff
@@ -388,6 +285,7 @@ function getLatestRichPresenceUpdates(): array
     return $playersFound;
 }
 
+// TODO dead code?
```
Comment (luchaos) [inline app/Helpers/database/user-activity.php:288]: open TODO

File: public/API/API_GetUserSummary.php [hunk 4]
```diff
@@ -104,24 +104,25 @@
...
+//    unset($retVal['LastActivityID']);
+// }
+//
+// $status = 'Offline';
+// if ($retVal['LastActivity']) {
+//    $lastUpdate = (int) date("U", strtotime($retVal['LastActivity']['lastupdate']));
+//    $now = (int) date("U");
+//    $status = ($lastUpdate + 600) > $now ? 'Online' : 'Offline';
+// }
+// $retVal['Status'] = $status;
```
Comment (luchaos) [inline public/API/API_GetUserSummary.php:126]: open TODO

File: public/dorequest.php [hunk 5]
```diff
@@ -237,15 +237,6 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
...
             PlayerSessionHeartbeat::dispatch($user, Game::find($gameID), $activityMessage);
 
-            // legacy rich presence support (deprecated - see ResumePlayerSession)
-            if (isset($activityMessage) && $user->LastGameID == $gameID) {
-                $user->RichPresenceMsg = utf8_sanitize($activityMessage);
-                $user->RichPresenceMsgDate = Carbon::now();
-            }
-
-            $user->LastLogin = Carbon::now();
-            $user->save();
```
Comment (luchaos) [inline public/dorequest.php:247]: last login update should be handled by PlayerSessionHeartbeat.
verify that this is actually the case.

File: public/dorequest.php [hunk 6]
```diff
@@ -303,6 +294,7 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
 
     case "postactivity":
         $activityType = (int) request()->input('a');
+        // TODO LegacyActivityType
         if ($activityType != ActivityType::StartedPlaying) {
```
Comment (luchaos) [inline public/dorequest.php:298]: keep this class, rename it?

File: public/history.php [hunk 7]
```diff
@@ -238,9 +236,11 @@ function convertDate() {
...
     echo "</b><br>";
 
-    $memberSince = !empty($userSignedUp) ? getNiceDate(strtotime($userSignedUp), true) : '';
-    echo "Member since: " . $memberSince . "<br>";
-    echo "<br>";
+    // TODO replace this with created date
+//    $memberSince = !empty($userSignedUp) ? getNiceDate(strtotime($userSignedUp), true) : '';
+//    echo "Member since: " . $memberSince . "<br>";
+//    echo "<br>";
+
```
Comment (luchaos) [inline public/history.php:243]: open TODO

File: public/usergameactivity.php [hunk 8]
```diff
@@ -3,6 +3,10 @@
 use App\Platform\Enums\AchievementFlag;
 use App\Site\Enums\Permissions;
 
+// TODO refactor to game_sessions or player_games
+
+return redirect(route('home'));
```
Comment (luchaos) [inline public/usergameactivity.php:8]: open TODO. addressed by Jamira's refactoring

File: tests/Feature/Api/V1/UserRecentlyPlayedGamesTest.php [hunk 9]
```diff
@@ -86,7 +67,7 @@ public function testGetUserRecentlyPlayedGames(): void
                     'ConsoleID' => $system->ID,
                     'ConsoleName' => $system->Name,
                     'ImageIcon' => $game->ImageIcon,
-                    'LastPlayed' => $activity2->lastupdate->__toString(),
+                    // 'LastPlayed' => $activity2->lastupdate->__toString(),
```
Comment (luchaos) [inline tests/Feature/Api/V1/UserRecentlyPlayedGamesTest.php:70]: what does LastPlayed actually refer to now

File: app/Helpers/database/user-activity.php (hunk 2 above)
Comment (Jamiras) [reply app/Helpers/database/user-activity.php:17]: Nope. That was previously removed (https://github.com/RetroAchievements/RAWeb/pull/1923/files#diff-4eb28129e711e17b32a464138654fc22cd98c92b7c74d5413a64cea0786e79b8L37). I'm guessing it got resurrected by a merge conflict.

File: app/Helpers/database/user-activity.php (hunk 3 above)
Comment (Jamiras) [reply app/Helpers/database/user-activity.php:287]: I think it used to be used for recently added achievements on the main page. Now we have recently added sets.

File: public/history.php (hunk 7 above)
Comment (Jamiras) [reply public/history.php:227]: `$userDetails->Created`. I almost made that change myself when working on this page yesterday.

File: public/usergameactivity.php (hunk 8 above)
Comment (Jamiras) [reply public/usergameactivity.php:8]: #1925

File: app/Helpers/database/ticket.php (hunk 1 above)
Comment (luchaos) [reply app/Helpers/database/ticket.php]: done in https://github.com/RetroAchievements/RAWeb/pull/1954

File: app/Platform/Actions/RevalidateAchievementSetBadgeEligibility.php [hunk 10]
```diff
@@ -120,17 +119,6 @@ private function revalidateCompletionBadgeEligibility(PlayerGame $playerGame): v
             );
             PlayerBadgeAwarded::dispatch($badge);
             PlayerGameCompleted::dispatch($playerGame->user, $playerGame->game);
-
-            // TODO WriteUserActivity
```
Comment (luchaos) [inline app/Platform/Actions/RevalidateAchievementSetBadgeEligibility.php:124]: let's not do that for now.
adding another listener for WriteUserActivity might be relevant for when activity feeds are implemented but it won't happen here, events are already dispatched.

File: public/history.php (hunk 7 above)
Comment (luchaos) [reply public/history.php]: done in https://github.com/RetroAchievements/RAWeb/pull/1937/commits/8496628ac0e070b3e6ebf57dd3b6e35fa478dc7a

**Review by Jamiras — APPROVED** [review]

**Pull request merged**